lib/classroom/models/assignment.rb

```diff
@@ -11,34 +11,47 @@ class Assignment
   embeds_many :submissions
 
   create_index({'organization': 1, 'course': 1, 'student.uid': 1})
+  create_index({'organization': 1, 'exercise.eid': 1, 'student.uid': 1})
   create_index({'organization': 1, 'course': 1, 'guide.slug': 1, 'exercise.eid': 1, 'student.uid': 1})
   create_index({'organization': 1, 'course': 1, 'guide.slug': 1, 'student.uid': 1, 'exercise.eid': 1})
   create_index({'guide.slug': 1, 'exercise.eid': 1}, {name: 'ExBibIdIndex'})
 
-  def comment!(comment, sid)
-    submissions.find_by!(sid: sid).comment! comment
+  def add_message!(message, sid)
```

Comment:
I'd rename this to `send_answer!` to keep consistent with laboratory's `send_question!` and `receive_answer!`